app/src/main/java/org/dhis2/usescases/datasets/dataSetTable/DataSetTableViewModelFactory.kt

```diff
@@ -8,6 +8,7 @@ import org.dhis2.commons.viewmodel.DispatcherProvider
 import org.dhis2.usescases.datasets.dataSetTable.DataSetTableContract.View
 import org.dhis2.utils.analytics.AnalyticsHelper
 
+@Suppress("UNCHECKED_CAST")
```

Comment:
Instead of ignoring the warning, what do you think about:
``
`


    override fun <T : ViewModel> create(modelClass: Class<T>): T {
        if (modelClass.isAssignableFrom(DataSetTablePresenter::class.java)) {
            return DataSetTablePresenter(
                view,
                dataSetTableRepository,
                periodUtils,
                dispatcherProvider,
                analyticsHelper,
                updateProcessor,
                openErrorSectionOnInit
            ) as T
        }
        throw IllegalArgumentException("Unknown ViewModel class: ${modelClass.name}")
    }
}
`

Reply:
This won't prevent the warning. It seems the best for viewmodels if version > 2.5.0 is to do this:
https://developer.android.com/topic/libraries/architecture/viewmodel/viewmodel-factories?hl=es-419